bindgen/src/context.ts

```diff
@@ -18,10 +18,11 @@
 
 import { Formatter } from "./formatter";
 import { Outputter } from "./outputter";
-import { Spec } from "./spec";
+import { OptInSpec, Spec } from "./spec";
 
 export type TemplateContext = {
   spec: Spec;
+  optInSpec?: OptInSpec;
```

Comment:
I wonder if we should change this to be like this instead?
```ts
rawSpec: Spec;
spec: BoundModel;
```

The reason that the "templates" got the raw `Spec` and not the `BoundModel`, so had to call `bindModel` themselves is mostly historical. I think all templates now call `bindModel()` so we should probably move that logic into the driver instead.

Feel free to say that you don't want to make that change as part of this PR, but if it is easy enough, it might be worth just doing it.

Reply:
I think that makes a lot of sense and simplifies the usage for the SDKs 🙂